src/main/java/org/springframework/data/elasticsearch/core/AbstractElasticsearchTemplate.java

```diff
@@ -868,4 +909,38 @@ public ReadSearchScrollDocumentResponseCallback(Class<T> type, IndexCoordinates
 		}
 	}
 	// endregion
+
+	// region routing
+	@Override
+	public void setRoutingResolver(RoutingResolver routingResolver) {
+
+		Assert.notNull(routingResolver, "routingResolver must not be null");
+
+		this.routingResolver = routingResolver;
+	}
+
+	@Override
+	public RoutingResolver getRoutingResolver() {
+		return Objects.requireNonNull(routingResolver);
+	}
+
+	@Override
+	public ElasticsearchOperations withRouting(RoutingResolver routingResolver) {
```

Comment:
Instead of additional methods, how about `RoutingResolver.just(String)`?

Reply:
done